Userland/Applications/FileManager/FileUtils.cpp

```diff
@@ -67,4 +68,91 @@ void delete_paths(const Vector<String>& paths, bool should_confirm, GUI::Window*
     }
 }
 
+void create_trash_bin()
+{
+    auto trash_bin_path = String::formatted("{}/.trash", getenv("HOME"));
+    auto trash_bin_files = String::formatted("{}/files", trash_bin_path);
+    auto trash_bin_info = String::formatted("{}/info", trash_bin_path);
+
+    if (Core::File::is_directory(trash_bin_path)) {
+        dbgln("Trash directory exists.");
+    } else {
+        if (mkdir(trash_bin_path.characters(), 0755) < 0) {
+            auto saved_errno = errno;
+            warnln("Failed to create directory {}: {}", trash_bin_path, strerror(saved_errno));
+        }
+    }
+
+    if (!Core::File::is_directory(trash_bin_files)) {
+        if (mkdir(trash_bin_files.characters(), 0755) < 0) {
+            auto saved_errno = errno;
+            warnln("Failed to create directory {}: {}", trash_bin_files, strerror(saved_errno));
+        }
+    }
+
+    if (!Core::File::is_directory(trash_bin_info)) {
+        if (mkdir(trash_bin_info.characters(), 0755) < 0) {
+            auto saved_errno = errno;
+            warnln("Failed to create directory {}: {}", trash_bin_info, strerror(saved_errno));
+        }
+    }
+}
+
+[[maybe_unused]] void move_to_trash(String const& path)
+{
+    [[maybe_unused]] auto file_info = "[Trash Info]\nPath={}\nDeletionDate={}\n";
+
+    auto file_name = LexicalPath(path).basename();
+    auto file_info_name = String::formatted("{}.trashinfo", file_name);
+    auto deletion_date = Core::DateTime::now().to_string("%Y-%m-%dT%H:%M:%S");
+
+    auto trash_bin_path = String::formatted("{}/.trash", getenv("HOME"));
```

Comment:
This will attempt to save in `/.trash` if the user's home directory is not set in the `$HOME` environment variable.

Please use `StandardPaths::home_directory()` instead:

https://github.com/SerenityOS/serenity/blob/b25eba10d2f60c54bcd3516183050b36bfa9f390/Userland/Libraries/LibCore/StandardPaths.cpp#L17-L26

Alternatively, it may be useful to add a `trash_directory` function to `LibCore::StandardPaths`.